schism/compiler.ss

```diff
@@ -1113,7 +1135,66 @@
                 (call %closure-free-var (var ,closure) (i32 ,index)))
 	      (bind-free-vars closure (cdr free-vars) (+ 1 index)))))
 
-  
+  (define (lower-literal x)
+    (cond
+     ((null? x)
+      `(call %get-null))
+     ((eq? x (void))
+      `(call %get-void))
+     ((number? x)
+      `(call %make-number (i32 ,x)))
+     ((boolean? x)
+      (if x `(call %get-true) `(call %get-false)))
+     ((char? x)
+      `(call %make-char (i32 ,(char->integer x))))
+     ((string? x)
+      `(call %list->string ,(lower-literal (string->list x))))
+     ((symbol? x)
+      `(call string->symbol ,(lower-literal (symbol->string x))))
```

Comment:
Incidentally, and I don't know why this is, if we change this to `%string->symbol` it goes 10x slower!!!

Strange considering that `string->symbol` is just a thin wrapper around `%string->symbol`, which is imported from the run-time.  Is there a performance penalty associated with the number of call-sites out to run-time routines, in V8?

Reply:
That's bizarre. I can't think of why V8 would have a penalty for the number of static call sites. Calling out to JS is moderately expensive, but this doesn't seem to affect the number of calls executed.

It could be that the length of call instruction for `%string->symbol` is longer than `symbol->string`, and this this is such a common call, the size of the Wasm bytes gets 10x larger, meaning it takes 10x longer to write it out. For example, if `%string->symbol`'s index were more than 127, it'd spill over into a 2-byte LEB, which could be the difference.

Anyway, that's my best guess. I'd be surprised to see that take 10x longer though.